resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1

```diff
@@ -812,6 +839,283 @@ class StickyKeys
+        $currentState.HotkeyActive = $currentFlags.HasFlag([ToggleKeysOptions]::HotkeyActive)
+        $currentState.ConfirmOnHotkeyActivation = $currentFlags.HasFlag([ToggleKeysOptions]::ConfirmHotkey)
+        $currentState.HotkeySound = $currentFlags.HasFlag([ToggleKeysOptions]::HotkeySound)
+        $currentState.VisualIndicator = $currentFlags.HasFlag([ToggleKeysOptions]::VisualIndicator)
+        
+        return $currentState
+    }
+
+    [bool] Test()
+    {
+        $currentState = $this.Get()
+
+        if (($null -ne $this.Active) -and ($this.Active -ne $currentState.Active))
+        {
+            return $false
+        }
+
+        if (($null -ne $this.Available) -and ($this.Available -ne $currentState.Available))
+        {
+            return $false
+        }
+
+        if (($null -ne $this.HotkeyActive) -and ($this.HotkeyActive -ne $currentState.HotkeyActive))
+        {
+            return $false
+        }
+
+        if (($null -ne $this.ConfirmOnHotkeyActivation) -and ($this.ConfirmOnHotkeyActivation -ne $currentState.ConfirmOnHotkeyActivation))
+        {
+            return $false
+        }
+
+        if (($null -ne $this.HotkeySound) -and ($this.HotkeySound -ne $currentState.HotkeySound))
+        {
+            return $false
+        }
+
+        if (($null -ne $this.VisualIndicator) -and ($this.VisualIndicator -ne $currentState.VisualIndicator))
+        {
+            return $false
+        }
+
+        return $true
+    }
+
+    [void] Set()
+    {
+        # Only make changes if changes are needed
+        if (-not $this.Test())
+        {
+            # If a value isn't set in the DSC, it should remain unchanged, to do this we need the current flags
+            $flags = [ToggleKeys]::GetCurrentFlags()
+
+            if ($null -ne $this.Active)
+            {
+                # If the user requested to set the value to true, and the current flag is unset, set the flag, otherwise add no flag
+                # If the user requested to set the value to false, and the current flag is set, unset the flag, otherwise remove no flag
+                # Since $this.Active can only be either true or false, only one of these statements will actually affect the value of $flags
+                $flags += ($this.Active -and !$flags.HasFlag([ToggleKeysOptions]::Active)) ? [ToggleKeysOptions]::Active : [ToggleKeysOptions]::None
+                $flags -= (!$this.Active -and $flags.HasFlag([ToggleKeysOptions]::Active)) ? [ToggleKeysOptions]::Active : [ToggleKeysOptions]::None
+            }
+
+            if ($null -ne $this.Available)
+            {
+                $flags += ($this.Available -and !$flags.HasFlag([ToggleKeysOptions]::Available)) ? [ToggleKeysOptions]::Available : [ToggleKeysOptions]::None
+                $flags -= (!$this.Available -and $flags.HasFlag([ToggleKeysOptions]::Available)) ? [ToggleKeysOptions]::Available : [ToggleKeysOptions]::None
+            }

```

Comment:
```suggestion
            if ($null -ne $this.Available)
            {
                if ($this.Available)
                {
                    $flags += !$flags.HasFlag([ToggleKeysOptions]::Available) ? [ToggleKeysOptions]::Available : [ToggleKeysOptions]::None
                }
                else
                {
                    $flags -= $flags.HasFlag([ToggleKeysOptions]::Available) ? [ToggleKeysOptions]::Available : [ToggleKeysOptions]::None
                }
            }

```

I feel that it is better if you separate these out so that only one 'flag set' is taking place rather than two (even if it's the same behavior).

Reply:
Is this purely for performance reasons, or readability, or just general structure preference?  I only ask because the flags aren't actually set until the very last line (969 for this specific resource), so the actual state of the flags on the machine isn't changing until the final desired state has been calculated, so even if there was a terminating error, at worst the machine would be in the same configuration it started in

Reply:
The main reason is readability but one could also argue performance as this avoids redundant condition checks and flag set/removals.